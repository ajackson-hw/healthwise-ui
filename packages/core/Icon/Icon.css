.Icon {
  width: 100%;
  height: 100%;
}

.IconSvg {
  display: block;
  width: 100%;
  height: 100%;
  fill: var(--color-text-primary);
}

.color_primary {
  fill: var(--color-text-primary);
}

.color_primary_stroke {
  stroke: var(--color-text-primary);
}

.color_contrast {
  fill: var(--color-primary-light);
}

.color_contrast_stroke {
  stroke: var(--color-primary-light);
}
