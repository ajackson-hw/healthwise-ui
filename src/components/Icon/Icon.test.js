/* global describe it expect */
import React from 'react'
import renderer from 'react-test-renderer'

import Icon, {
  AddIcon,
  AnalyticsIcon,
  AnalyticsSimpleIcon,
  BrandingIcon,
  BrandingSimpleIcon,
  CheckboxCheckedIcon,
  CheckboxUncheckedIcon,
  CheckIcon,
  CollapseIcon,
  DashboardIcon,
  DownArrowIcon,
  DownloadIcon,
  EditIcon,
  ExpandIcon,
  GroupsIcon,
  HomeIcon,
  InfoIcon,
  KeyIcon,
  LeftArrowIcon,
  LockedIcon,
  MagnifyingGlassIcon,
  PrinterIcon,
  ProfileIcon,
  RestoreIcon,
  RightArrowIcon,
  TrashIcon,
  UnlockedIcon,
  UpArrowIcon,
  UsersIcon,
  UsersSimpleIcon,
  WarningIcon,
} from './index'

const simpleSvg = <path d="M16.6 8L12 13 7.4 8 6 9.5l6 6.5 6-6.5" fillRule="evenodd" />
const complexSvg = (
  <g fill="none" fillRule="evenodd">
    <path fill="#FCBC3F" d="M41 142h26V84H41m34 58h26V39H75m37.7 103.4h26v-76h-26" />
    <path
      stroke="#FFF"
      strokeWidth="4"
      d="M76 132h15v-27H76zm56 0h15V61h-15zm-28 0h15v-21h-15zm-56 0h15v-11H48zm83-85V26h-21m20.8.2l-92.8 94M20 132h11m-11-12h11m-11-12h11M20 96h11M20 85h11M20 73h11M20 61h11M20 50h11M20 38h11M20 144h130"
    />
  </g>
)

describe('<Icon />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Icon>{simpleSvg}</Icon>)
    expect(tree).toMatchSnapshot()
  })

  it('renders all icons correctly', () => {
    const tree = renderer.create(
      <div>
        <AddIcon />
        <AnalyticsIcon />
        <AnalyticsSimpleIcon />
        <BrandingIcon />
        <BrandingSimpleIcon />
        <CheckboxCheckedIcon />
        <CheckboxUncheckedIcon />
        <CheckIcon />
        <CollapseIcon />
        <DashboardIcon />
        <DownArrowIcon />
        <DownloadIcon />
        <EditIcon />
        <ExpandIcon />
        <GroupsIcon />
        <HomeIcon />
        <InfoIcon />
        <KeyIcon />
        <LeftArrowIcon />
        <LockedIcon />
        <MagnifyingGlassIcon />
        <PrinterIcon />
        <ProfileIcon />
        <RestoreIcon />
        <RightArrowIcon />
        <TrashIcon />
        <UnlockedIcon />
        <UpArrowIcon />
        <UsersIcon />
        <UsersSimpleIcon />
        <WarningIcon />
      </div>
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with viewBox property', () => {
    const tree = renderer.create(<Icon viewBox="0 0 170 170">{complexSvg}</Icon>)
    expect(tree).toMatchSnapshot()
  })
})
