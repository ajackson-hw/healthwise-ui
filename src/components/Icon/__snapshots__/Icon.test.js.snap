// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders all icons correctly 1`] = `
<div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zm-2.6 13.3h-5.3v5.3h-2.7v-5.3H5.3v-2.7h5.3V5.3h2.7v5.3h5.3v2.7h.1z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 96 96"
    >
      <path
        className="color_contrast"
        d="M72.76 38.94h14.17v50.33H72.76zm-58.55 33.5h13.81v16.71H14.21zm19.68-13.49h14v30.33h-14zm19.44 5.45h13.94v24.88H53.33z"
      />
      <path
        className="color_primary"
        d="M9.83 81.71H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm0-9.13H2.42a.87.87 0 1 1 0-1.73h7.41a.87.87 0 1 1 0 1.73zm82.76 81.69H2.42a.84.84 0 1 1 0-1.67h90.17a.84.84 0 1 1 0 1.67zM23.55 57.28a1 1 0 0 1-.81-.42 1 1 0 0 1 .24-1.4l58.87-41.75A1 1 0 0 1 83 15.34L24.13 57.09a1 1 0 0 1-.58.19z"
      />
      <path
        className="color_primary"
        d="M83.51 23.28a1 1 0 0 1-1-1v-7.94h-8.92a1 1 0 0 1 0-2h9.91a1 1 0 0 1 1 1v8.93a1 1 0 0 1-1 1z"
      />
      <path
        className="color_primary_stroke"
        d="M76.71 35.44H90.3v50.5H76.71zM17.64 69.18h13.59v16.76H17.64zm19.69-13.66h13.59v30.43H37.33zm19.69 5.46h13.59v24.96H57.02z"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.72"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.1.06v23.9H24v-2.7H2.6V.06H.1zm20.2 19.4h3.5V3.76h-3.5v15.7zm-10.7 0h3.5V5.56H9.6v13.9zm5.2 0h3.5V8.96h-3.5v10.5zm-10.5 0h3.5v-7H4.3v7z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 96 96"
    >
      <rect
        className="color_primary_stroke"
        fill="none"
        height="59"
        rx="2.88"
        ry="2.88"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
        width="90.67"
        x="2.64"
        y="7.36"
      />
      <path
        className="color_contrast"
        d="M31.06 25.81l8.31 9.9 12.37-20.15 22.09 38-60.04.06 17.27-27.81z"
      />
      <rect
        className="color_primary_stroke"
        fill="none"
        height="5.83"
        rx="2.57"
        ry="2.57"
        strokeWidth="1.5"
        width="39.68"
        x="28.4"
        y="84.69"
      />
      <path
        className="color_primary_stroke"
        d="M28.72 85.94l7.83-19.7m23.34 0l7.75 19.54"
        fill="none"
        strokeMiterlimit="10"
        strokeWidth="1.5"
      />
      <path
        className="color_primary_stroke"
        d="M37.73 29.31l8.3 9.9 12.38-20.15 22.09 38-60.04.06 17.27-27.81z"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M23.3 1.4H.7c-.3 0-.5.2-.5.5v15.7c0 .3.2.5.5.5h8.2L8.2 21H8c-.3 0-.5.4-.5.9s.2.9.5.9h7.9c.3 0 .5-.4.5-.9s-.2-.9-.5-.9h-.2l-.6-2.9h8.2c.3 0 .5-.2.5-.5V1.9c0-.3-.3-.5-.5-.5zM9.2 21l.6-2.9h4.3l.6 2.9H9.2zM8.7 6.6l2.2 2.6 3.2-5.3 5.8 9.9H4.3l4.4-7.2z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zM10.8 17.4l-2.1 2.1-6.2-6.2 2.1-2.1 4.1 4.1L19.4 4.5l2.1 2.1-10.7 10.8z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.3 2.7v18.7H2.7V2.7h18.6zm0-2.7H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.9 16.9l-5.4-5.3-1.8 1.8 7.2 7.1L23.3 5.3l-1.8-1.8z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zm0 21.3H2.7V2.7h18.7v18.6h-.1z"
      />
      <path
        d="M12 5.3l-5.3 5.4h4V20h2.6v-9.3h4"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 10a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm-1.26-4.703a1.012 1.012 0 0 0-1.435 0 1.004 1.004 0 0 0 0 1.444 1.02 1.02 0 0 0 1.434-1.444zM14 15.755a1.982 1.982 0 0 0-.368-1.132L11.494 8l-2.145 6.677c-.194.29-.313.638-.324 1.01L9 15.764h.012v.01c0 1.229 1.116 2.225 2.494 2.225C12.884 18 14 17.004 14 15.774v-.019zM11 3a1 1 0 1 0 0 2 1 1 0 0 0 0-2zM6.701 5.293A.992.992 0 0 0 5.997 5 .999.999 0 0 0 5 6c0 .552.446 1 .997 1a.992.992 0 0 0 .704-.293.983.983 0 0 0 0-1.414zM4 10a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm8 14C5.372 24 0 18.628 0 12S5.372 0 12 0s12 5.372 12 12-5.372 12-12 12z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon hw-icon-down-arrow"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.2 4L12 13.9 2.8 4 0 7l12 13L24 7"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zm0 16H16c0 2.2-1.8 4-4 4s-4-1.8-4-4H2.7V2.7h18.7V16h-.1zm-4-6.7h-2.7v-4H9.3v4H6.7l5.3 5.3 5.3-5.3z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14.6 5.4L16 6.8 5.4 17.4l-.4-1-1.2-.2L14.6 5.4zM2.8 17.3l1.4.2.7 1.6 1.8.8.1 1.3-.1.1-2.8.4-1.5-1.5.4-2.8v-.1zm5 2.8l-.1-.9-1.2-.6L17.2 8l1.4 1.4L7.8 20.1zm6.8-16.9l-1.1 1.1L1.1 16.7 0 24l7.3-1.1 12.4-12.4 1.1-1.1-6.2-6.2zM15.9 2L22 8.1l1.4-1.4c.7-.7.7-1.9 0-2.6L19.8.5c-.7-.7-1.9-.7-2.6 0L15.9 2z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.3 0H2.7C1.2 0 0 1.2 0 2.7v18.7C0 22.8 1.2 24 2.7 24h18.7c1.5 0 2.7-1.2 2.7-2.7V2.7C24 1.2 22.8 0 21.3 0zm0 21.3H2.7V2.7h18.7v18.6h-.1z"
      />
      <path
        d="M13.3 14.7V5.3h-2.6v9.4h-4L12 20l5.3-5.3"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 170 170"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          className="color_contrast"
          d="M61 49c0 11.6-9.4 21-21 21s-21-9.4-21-21 9.4-21 21-21 21 9.4 21 21m49 84.48c0 12.42-10.08 22.5-22.5 22.5-12.43 0-22.5-10.08-22.5-22.5 0-12.43 10.07-22.5 22.5-22.5 12.42 0 22.5 10.07 22.5 22.5"
        />
        <path
          className="color_contrast"
          d="M150.5 50.5c0 8.3-6.72 15-15 15-8.3 0-15-6.7-15-15 0-8.28 6.7-15 15-15 8.28 0 15 6.72 15 15"
        />
        <path
          className="color_primary_stroke"
          d="M39.5 99.94V57.2m42.16 73.97l-35.4-20.8m83.54-.47l-36.18 21.27m42.14-73.82v42.27M93.63 25.04l35.3 20.76m-82.98.22l35.27-20.74M66.04 65.4l-19.9-11.7m83.42 49.04l-18.3-10.76m.22-27.9l17.9-10.52m-83.52 49.1L65.68 91M88 50.66V28m0 98.66V104M47.55 50.3c0 3.85-3.14 7-7 7-3.87 0-7-3.15-7-7 0-3.88 3.13-7 7-7 3.86 0 7 3.12 7 7zm94.17-.16c0 3.8-3.02 6.86-6.73 6.86-3.76 0-6.77-3.07-6.77-6.86 0-3.78 3-6.85 6.74-6.85 3.7 0 6.72 3.04 6.72 6.82zM95 22c0 3.87-3.14 7-7 7-3.87 0-7-3.13-7-7s3.13-7 7-7c3.86 0 7 3.13 7 7zm-.53 111.86c0 3.78-3.02 6.85-6.74 6.85-3.72.04-6.73-3.04-6.73-6.82 0-3.8 3-6.86 6.73-6.86 3.72 0 6.74 3.08 6.74 6.87zm-46.92-27c0 3.86-3.14 7-7 7-3.87 0-7-3.14-7-7 0-3.87 3.13-7 7-7 3.86 0 7 3.13 7 7zm94.7 0c0 3.86-3.14 7-7 7-3.87 0-7-3.14-7-7 0-3.87 3.13-7 7-7 3.86 0 7 3.13 7 7zM115 79c0 14.9-12.1 27-27 27S61 93.9 61 79s12.1-27 27-27 27 12.1 27 27z"
          strokeWidth="4"
        />
        <path
          className="color_primary_stroke"
          d="M97.72 71.8c0 5.14-4.1 9.3-9.14 9.3-5.05 0-9.15-4.16-9.15-9.3 0-5.15 4.1-9.32 9.15-9.32 5.05 0 9.14 4.17 9.14 9.3z"
          strokeWidth="4"
        />
        <path
          className="color_primary_stroke"
          d="M106.58 98v-8.18c0-2.6-2-5.6-4.56-6.4-8.77-2.67-18.12-2.67-26.9 0-2.54.8-4.54 3.8-4.55 6.4l-.08 8.82m8.33-7.24v12.18M98.3 91.4v12.18"
          strokeWidth="4"
        />
      </g>
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.5 24V12h-15v12H9v-9h6v9h4.5zM12 0L1.5 8.6V12h21V8.6L12 0z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 12c0 6.6 5.4 12 12 12s12-5.4 12-12S18.6 0 12 0 0 5.4 0 12zm13.2 6h-2.4v-2.4h2.4V18zm0-4.8h-2.4V6h2.4v7.2z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12.7 9.8c-.9-2.5-3.3-4.4-6.2-4.4C2.9 5.5 0 8.4 0 12s2.9 6.5 6.5 6.5c2.8 0 5.3-1.8 6.2-4.4h4.7v4.4h4.4v-4.4H24V9.8H12.7zm-6.2 4.4c-1.2 0-2.2-1-2.2-2.2s1-2.2 2.2-2.2 2.2 1 2.2 2.2-1 2.2-2.2 2.2z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 21.2L10.1 12 20 2.8 17 0 4 12l13 12"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.9 8.2h-1.1V4.5c.1-2.5-2-4.5-4.5-4.5h-4.2C7.6 0 5.5 2 5.5 4.5v3.7H4.1c-1.2 0-2.1.9-2.1 2.1V22c0 1.1.9 2 2.1 2h15.8c1.1 0 2.1-.9 2.1-2.1V10.3c0-1.2-.9-2.1-2.1-2.1zM7.8 8.1V4.5c0-1.2 1-2.3 2.3-2.3h4.2c1.3 0 2.3 1 2.3 2.3v3.7l-8.8-.1zm11.9 13.6H4.3V10.4h15.4v11.3z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17.2 15.1h-1.1l-.4-.4c1.3-1.6 2.2-3.6 2.2-5.8C17.9 4 13.9 0 9 0S0 4 0 8.9s4 8.9 8.9 8.9c2.2 0 4.2-.8 5.8-2.2l.4.4v1.1L22 24l2-2-6.8-6.9zm-8.3 0c-3.4 0-6.2-2.8-6.2-6.2s2.8-6.2 6.2-6.2 6.2 2.8 6.2 6.2-2.8 6.2-6.2 6.2z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon hw-icon-printer"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.4 7.2H3.6c-2 0-3.6 1.6-3.6 3.6V18h4.8v4.8h14.4V18H24v-7.2c0-2-1.6-3.6-3.6-3.6zm-3.6 13.2H7.2v-6h9.6v6zm3.6-8.4c-.7 0-1.2-.5-1.2-1.2s.5-1.2 1.2-1.2 1.2.5 1.2 1.2-.5 1.2-1.2 1.2zM19.2 1.2H4.8V6h14.4V1.2z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 15c-4 0-12 2-12 6v3h24v-3c0-4-8-6-12-6zM12 12c3.3 0 6-2.7 6-6s-2.7-6-6-6-6 2.7-6 6 2.7 6 6 6z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13.7 9.7c-1.3 0-2.3 1-2.3 2.3s1 2.3 2.3 2.3 2.3-1 2.3-2.3-1-2.3-2.3-2.3zm0-8C8 1.7 3.4 6.3 3.4 12H0l4.6 4.6L9.1 12H5.7c0-4.4 3.6-8 8-8s8 3.6 8 8-3.6 8-8 8c-1.7 0-3.3-.6-4.6-1.5l-1.6 1.6c1.7 1.3 3.9 2.1 6.3 2.1C19.4 22.3 24 17.7 24 12S19.4 1.7 13.7 1.7z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4 2.8l9.9 9.2L4 21.2 7 24l13-12L7 0"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M2.8 7.4h18.4V3.7H2.8"
      />
      <path
        d="M17.3 20.2c0 .5-.4.9-.9.9s-.9-.4-.9-.9V9.5c0-.5.4-.9.9-.9s.9.4.9.9v10.7zm-4.4 0c0 .5-.4.9-.9.9s-.9-.4-.9-.9V9.5c0-.5.4-.9.9-.9s.9.4.9.9v10.7zm-4.4 0c0 .5-.4.9-.9.9s-.9-.4-.9-.9V9.5c0-.5.4-.9.9-.9s.9.4.9.9v10.7zM4 6.8V23c0 .6.4 1 1 1h14c.5 0 1-.5 1-1V6.8H4zm13.5-5.5L16.2 0H7.8L6.5 1.3l1.6 1.2.9-.9h6l.9.9"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.9 8.2H7.8V4.5c0-1.2 1-2.3 2.3-2.3h4.2c1.3 0 2.3 1 2.3 2.3v.2c0 .6.5 1.1 1.1 1.1.6 0 1.1-.5 1.1-1.1v-.2c.1-2.5-2-4.5-4.5-4.5H10C7.5 0 5.5 2 5.5 4.5v3.7H4.1c-1.2 0-2.1.9-2.1 2.1V22c0 1.1.9 2 2.1 2h15.8c1.1 0 2.1-.9 2.1-2.1V10.3c0-1.2-.9-2.1-2.1-2.1zm-.2 13.5H4.3V10.4h15.4v11.3z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon hw-icon-up-arrow"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.2 20L12 10.1 2.8 20 0 17 12 4l12 13"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 96 96"
    >
      <circle
        className="color_contrast"
        cx="81.27"
        cy="67.13"
        r="11.96"
      />
      <circle
        className="color_contrast"
        cx="13.96"
        cy="28.31"
        r="11.96"
      />
      <circle
        className="color_contrast"
        cx="81.19"
        cy="28.31"
        r="11.96"
      />
      <path
        className="color_primary"
        d="M14.1 34.73a6.14 6.14 0 1 1 6.14-6.14 6.14 6.14 0 0 1-6.14 6.14zm0-10.54a4.41 4.41 0 1 0 4.41 4.4 4.41 4.41 0 0 0-4.41-4.4zm66.81 10.54A6.14 6.14 0 1 1 87 28.59a6.14 6.14 0 0 1-6.09 6.14zm0-10.54a4.41 4.41 0 1 0 4.4 4.4 4.42 4.42 0 0 0-4.4-4.4zM14.1 73.12A6.14 6.14 0 1 1 20.24 67a6.14 6.14 0 0 1-6.14 6.12zm0-10.54A4.4 4.4 0 1 0 18.51 67a4.4 4.4 0 0 0-4.41-4.42zm66.81 10.54A6.14 6.14 0 1 1 87 67a6.14 6.14 0 0 1-6.09 6.12zm0-10.54a4.4 4.4 0 1 0 4.4 4.4 4.41 4.41 0 0 0-4.4-4.4zM47.29 30.6a.87.87 0 0 1-.87-.87V14.35a.87.87 0 1 1 1.74 0v15.38a.87.87 0 0 1-.87.87zm0 51.14a.87.87 0 0 1-.87-.87V65.49a.87.87 0 1 1 1.74 0v15.38a.87.87 0 0 1-.87.87z"
      />
      <path
        className="color_primary"
        d="M31.92 39.35a.88.88 0 0 1-.44-.12l-13-7.61a.86.86 0 1 1 .87-1.49l13 7.6a.88.88 0 0 1 .31 1.19.86.86 0 0 1-.74.43zm44.15 26.24a.8.8 0 0 1-.44-.13L62.4 57.61a.88.88 0 0 1-.3-1.19.86.86 0 0 1 1.19-.3L76.52 64a.87.87 0 0 1-.45 1.62zm-56.98-.09a.85.85 0 0 1-.74-.42.86.86 0 0 1 .29-1.19l12.83-7.75a.87.87 0 0 1 .9 1.48l-12.83 7.75a.84.84 0 0 1-.45.13zm43.73-26.07a.87.87 0 0 1-.45-1.61L75.44 30a.88.88 0 0 1 1.19.3.86.86 0 0 1-.3 1.19l-13.07 7.8a.88.88 0 0 1-.44.14zM42.73 85a.82.82 0 0 1-.43-.12L18.08 70.79a.87.87 0 0 1 .87-1.5L43.17 83.4a.86.86 0 0 1 .31 1.18.86.86 0 0 1-.75.42zM18.52 26.47a.86.86 0 0 1-.75-.43.88.88 0 0 1 .31-1.19l24.22-14.1a.87.87 0 0 1 .87 1.5L19 26.35a.82.82 0 0 1-.48.12zM14 62.22a.87.87 0 0 1-.87-.86V34a.87.87 0 0 1 1.74 0v27.36a.87.87 0 0 1-.87.86zM52.56 85a.86.86 0 0 1-.75-.43.86.86 0 0 1 .31-1.18L76.19 69.3a.87.87 0 1 1 .88 1.49L53 84.89a.88.88 0 0 1-.44.11zm23.93-58.67a.88.88 0 0 1-.44-.12l-23.93-14a.87.87 0 0 1 .88-1.5l23.93 14a.87.87 0 0 1-.44 1.62zm4.84 35.89a.86.86 0 0 1-.86-.86V34a.87.87 0 1 1 1.73 0v27.36a.87.87 0 0 1-.87.86z"
      />
      <path
        className="color_primary"
        d="M47.66 15.22a6.14 6.14 0 1 1 6.14-6.14 6.14 6.14 0 0 1-6.14 6.14zm0-10.54a4.4 4.4 0 1 0 4.41 4.4 4.41 4.41 0 0 0-4.41-4.4zm0 87.6a6.14 6.14 0 1 1 6.14-6.14 6.15 6.15 0 0 1-6.14 6.14zm0-10.54a4.4 4.4 0 1 0 4.41 4.4 4.41 4.41 0 0 0-4.41-4.4zm0-15.67a18.53 18.53 0 1 1 18.53-18.53 18.55 18.55 0 0 1-18.53 18.53zm0-35.32a16.79 16.79 0 1 0 16.8 16.79 16.81 16.81 0 0 0-16.8-16.79z"
      />
      <circle
        className="color_primary_stroke"
        cx="47.51"
        cy="41.69"
        fill="none"
        r="6.49"
        strokeMiterlimit="10"
        strokeWidth="1.75"
      />
      <path
        className="color_primary_stroke"
        d="M47.22 65.17a18 18 0 0 1-11.48-4.6c-.5-.41-.67-.69-.56-1.16 1.64-7.22 5.46-10.81 12.33-10.81s10.76 4 12.39 11.16c.1.43 0 .52-.19.71a17.81 17.81 0 0 1-12.49 4.7z"
        fill="none"
        strokeMiterlimit="10"
        strokeWidth="1.5"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <circle
        cx="12"
        cy="7.1"
        r="3.2"
      />
      <circle
        cx="19.1"
        cy="7.8"
        r="2.4"
      />
      <path
        d="M23.7 15.7c-.5-3-2-4.7-4.7-4.7-1 0-1.9.2-2.5.7l.3.2a10 10 0 0 1 2.3 5.1v.2c2.6.1 4.8-.3 4.6-1.5z"
      />
      <circle
        cx="4.9"
        cy="7.8"
        r="2.4"
      />
      <path
        d="M7.2 12l.2-.2c-.6-.6-1.5-.8-2.5-.8-2.6 0-4.2 1.7-4.7 4.7-.2 1.2 2.1 1.6 4.7 1.6v-.2A9.3 9.3 0 0 1 7.2 12zm4.8 7.4c-3.5 0-6.5-.6-6.3-2.1.7-4 2.7-6.3 6.3-6.3s5.6 2.4 6.3 6.3c.2 1.4-2.9 2.1-6.3 2.1z"
      />
    </svg>
  </div>
  <div
    className="hw-icon Icon"
  >
    <svg
      className="IconSvg"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 22.4h24L12 1.6 0 22.4zm13.1-3.3h-2.2v-2.2h2.2v2.2zm0-4.4h-2.2v-4.4h2.2v4.4z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Icon /> renders correctly 1`] = `
<div
  className="hw-icon Icon"
>
  <svg
    className="IconSvg"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.6 8L12 13 7.4 8 6 9.5l6 6.5 6-6.5"
      fillRule="evenodd"
    />
  </svg>
</div>
`;

exports[`<Icon /> renders correctly with viewBox property 1`] = `
<div
  className="hw-icon Icon"
>
  <svg
    className="IconSvg"
    focusable="false"
    viewBox="0 0 170 170"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        d="M41 142h26V84H41m34 58h26V39H75m37.7 103.4h26v-76h-26"
        fill="#FCBC3F"
      />
      <path
        d="M76 132h15v-27H76zm56 0h15V61h-15zm-28 0h15v-21h-15zm-56 0h15v-11H48zm83-85V26h-21m20.8.2l-92.8 94M20 132h11m-11-12h11m-11-12h11M20 96h11M20 85h11M20 73h11M20 61h11M20 50h11M20 38h11M20 144h130"
        stroke="#FFF"
        strokeWidth="4"
      />
    </g>
  </svg>
</div>
`;
